pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -108,23 +157,44 @@ func (spc *realStatefulPodControl) UpdateStatefulPod(set *apps.StatefulSet, pod
 				return err
 			}
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetAutoDeletePVC) {
+			// if the Pod's PVCs are not consistent with the StatefulSet's PVC deletion policy, update the PVC
+			// and dirty the pod.
+			if match, err := spc.ClaimsMatchRetentionPolicy(set, pod); err != nil {
+				if consistent {
+					// Oops, something has gone badly wrong.
+					spc.recordPodEvent("update", set, pod, err)
+					return err
+				}
+				// Otherwise the error may be due to some other update, and we'll have to reconcile the next time around.
+				// TODO: does the reconcile need to be sped up?
```

Comment:
TODO no, if you fail on one loop you just try again on the next (any error should requeue you).  So I think you can remove this TODO

Reply:
Done